### PR TITLE
DM-21373: Add an internal timeout counter to the spectrograph polling

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

**Comment by parejkoj** [comment]
Note to self: the CSC should probably catch this and change the ExposureState to `TIMEOUT`.

File: tests/test_avsFiberSpectrograph.py [hunk 1]
```diff
@@ -424,6 +424,24 @@ def test_get_status_getAnalogIn_fails(self):
...
+    async def test_expose_PollScan_timeout(self):
+        """Test that `expose` raises if it has to wait too long when polling.
+        """
+        duration = 0.5  # seconds
+        # Have the PollScan just run forever.
+        self.patch.return_value.AVS_PollScan.side_effect = itertools.repeat(0)
+
+        spec = AvsFiberSpectrograph()
+        # asyncio.TimeoutError would be raised if the `wait_for` times out,
+        # but the message would not include this text.
```
Comment (r-owen) [inline tests/test_avsFiberSpectrograph.py]: Thank you for the helpful comment.

File: tests/test_csc.py [hunk 2]
```diff
@@ -214,6 +214,25 @@ def setUp(self):
...
+        # Have the PollScan just run forever.
+        self.patch.return_value.AVS_PollScan.side_effect = itertools.repeat(0)
+
+        async with Harness(initial_state=salobj.State.ENABLED) as harness:
+            # Check that we are properly in ENABLED at the start.
+            await self.check_summaryState(harness.remote, salobj.State.ENABLED)
+
+            msg = "Timeout waiting for exposure"
+            with salobj.assertRaisesAckError(ack=salobj.SalRetCode.CMD_FAILED, result_contains=msg):
+                await harness.remote.cmd_expose.set_start(timeout=STD_TIMEOUT, duration=.1)
```
Comment (r-owen) [inline tests/test_csc.py]: I strongly suggest `timeout=STD_TIMEOUT+duration`

Comment (parejkoj) [reply tests/test_csc.py]: Good catch, thanks.

File: python/lsst/ts/FiberSpectrograph/avsFiberSpectrograph.py [hunk 3]
```diff
@@ -405,14 +414,19 @@ def check_expose_ok(self, duration):
         measure_sleep = asyncio.create_task(asyncio.sleep(duration))
         await measure_sleep
 
+        start_time = time.perf_counter()
         data_available = 0
         while data_available != 1:
             self.log.debug("Polling for measurement.")
             data_available = self.libavs.AVS_PollScan(self.handle)
             assert_avs_code(data_available, "PollScan")
+            if (time.perf_counter() - start_time) > self.pollscan_timeout:
```
[the remark this answers is not in the document]
Comment (parejkoj) [reply python/lsst/ts/FiberSpectrograph/avsFiberSpectrograph.py]: Ah, the joys of multi-system clock support... After some digging, it looks like on non-windows architectures,` perf_counter` uses `monotonic` underneath.

The fact that so many mangled clock things are due to windows having had such terrible time support for so many years (e.g. not supporting BIOS UTC clocks) brings me perpetual joy.

**Pull request merged**